File: Sistema_de_Salarios/Salarios_jv/src/Funcionario.java
public abstract class Funcionario {

    //ATRIBUTOS
    public String nome;
    public String cpf;
    protected double salario_base;

    //CONSTRUTOR
    public Funcionario(String nome_input, String cpf_input) throws CPF_Invalido{
        nome = nome_input;
        cpf = cpf_input;
        if(!verificaCPF(cpf)) throw new CPF_Invalido("O CPF inserido é inválido\n");
        salario_base = 2000;
    }

    //MÉTODOS DE ACESSO
    protected double get_salario_base(){
        return salario_base;
    }
    public String get_nome(){
        return nome;
    }
    public String get_CPF(){
        return cpf;
    }

    //MÉTODO DE IMPRESSÃO

    @Override
    public String toString() {
        return String.format("Nome: %s\nCPF :%s\nSalário: %f\n", get_nome(), get_CPF(), calculaSalario());
    }

    //MÉTODOS ABSTRATOS
    protected abstract double calculaSalario();

    //MÉTODOS ESTÁTICOS
    static boolean verificaCPF(String cpf){

        int[] multiplicadores = {11, 10, 9, 8, 7, 6, 5, 4, 3, 2};
        int soma = 0;
        int valor_char;
        int resto;

        //VERIFICAÇÃO DO PRIMEIRO DÍGITO DE CONTROLE

        // primeira interação: realiza a soma
        for(int i = 0;  i < 9;  i++){
            valor_char = Character.digit(cpf.charAt(i), 10);

            soma = soma + valor_char * (multiplicadores[i] - 1);
        }

        // verifica o módulo
        resto = (soma*10) % 11;
        if(resto == 10) resto = 0;

        if(resto != Character.digit(cpf.charAt(9), 10)) return false;

        //VERIFICAÇÃO DO SEGUNDO DÍGITO DE CONTROLE
        soma = 0;

        // primeira interação: realiza a soma incluindo o primeiro dígito de controle
        for(int i = 0;  i < 10;  i++){
            valor_char = Character.digit(cpf.charAt(i), 10);
            soma = soma + valor_char * (multiplicadores[i]);

        }

        // verifica o módulo
        resto = (soma*10) % 11;
        if(resto == 10) resto = 0;

        //FINALIZAÇÃO
        return resto == Character.digit(cpf.charAt(10), 10);
    }

}

File: Sistema_de_Salarios/Salarios_py/Funcionario.py
from abc import ABC, abstractmethod


# -> CLASSE DE EXCESSÃO
class CPF_invalido(Exception):
    pass


# -> SUPER CLASSE ABSTRATA
class Funcionario(ABC):

    # CONSTRUTOR
    def __init__(self, nome, cpf):
        self.nome = nome
        self.cpf = cpf
        if not self.verifica_cpf(cpf):
            raise CPF_invalido("CPF inserido não é válido\n")
        self._salarioBase = 2000

    @abstractmethod
    def calcula_salario(self):
        pass

    # ACESSORES
    def get_nome(self):
        return self.nome

    def get_cpf(self):
        return self.cpf

    def _get_salario(self):
        return self._salarioBase

    # IMPRESSÃO
    def __str__(self):
        returned_string = f"Nome:{self.nome}\nCPF:{self.cpf}\nSalário:{self.calcula_salario()}\n"
        return returned_string

    # VERIFICADOR DE CPF

    @staticmethod
    def verifica_cpf(cpf):
        multiplicadores = [11, 10, 9, 8, 7, 6, 5, 4, 3, 2]
        soma = 0

        # Verificação do primeiro dígito de controle;
        for i in range(0, 9):
            soma = soma + int(cpf[i]) * (multiplicadores[i] - 1)

        resto = (soma * 10) % 11
        if resto == 10:
            resto = 0

        if resto != int(cpf[9]):
            return False

        # Verificação do segundo dígito de controle:

        soma = 0

        for i in range(0, 10):
            soma = soma + int(cpf[i]) * multiplicadores[i]

        resto = (soma * 10) % 11
        if resto == 10:
            resto = 0

        if resto != int(cpf[10]):
            return False
        else:
            return True


# -> SUB CLASSES

class Gerente(Funcionario):

    # CONSTRUTOR
    def __init__(self, nome, cpf):
        super().__init__(nome, cpf)
        self.multiplicador_salario = 2

    # OVER RIDE
    def calcula_salario(self):
        return self.multiplicador_salario * self._get_salario()


class Assistente(Funcionario):

    # CONSTRUTOR
    def __init__(self, nome, cpf):
        super().__init__(nome, cpf)
        self.multiplicador_salario = 1

    # OVER RIDE
    def calcula_salario(self):
        return self.multiplicador_salario * self._get_salario()


class Vendedor(Funcionario):

    # CONSTRUTOR
    def __init__(self, nome, cpf, comissao_input):
        super().__init__(nome, cpf)
        self.multiplicador_salario = 1
        self.comissao = comissao_input

    # OVER RIDE
    def calcula_salario(self):
        return self.multiplicador_salario * self._get_salario() + self.comissao
